blocks/library/shortcode/index.js

```diff
@@ -2,14 +2,143 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { withInstanceId, Dashicon } from '@wordpress/components';
+import { withInstanceId, Dashicon, Spinner, SandBox } from '@wordpress/components';
+import { Component, compose } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+import BlockControls from '../../block-controls';
+import { getCurrentPostId } from '../../../editor/store/selectors';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import './editor.scss';
 import PlainText from '../../plain-text';
 
+export class Shortcode extends Component {
```

Comment:
Having a hard time figuring out everything here :) do you think we can separate the rendering of the preview into a separate component `ShortcodePreview` taking the content and the post id as a prop and showing the preview? It could use `withAPIData` to fetch the preview.

Reply:
Thanks for the feedback @youknowriad . Sure, will try to refactor this piece to a separate component. 
I was just looking up the `withAPIData` component, and at how to send query params ( since I need to send the shortcode and postId to the REST API). The <a href="https://github.com/WordPress/gutenberg/blob/master/blocks/library/latest-posts/block.js" target="_blank">latest-posts block </a>, which uses this, seems to have a problem. I get a 404 error. Do you see this on your build too? (I have rebased against master to have the latest version).  Attaching below screenshot of what my Chrome network dev tool sees:
![chrome](https://user-images.githubusercontent.com/1269602/35825149-f6ec5372-0ada-11e8-8faa-edc43b0c02f9.png)

I'm assuming this to not be a bug with `withAPIData` component, but nevertheless bringing to your attention. The extra `?` seems to be an issue?

Reply:
Sometimes issues like that happen if you use "plain links" for permalinks, try using something else.

Reply:
@youknowriad Yes, issue was with plain links. Changed permalinks in settings and it works now. 
I've refactored code, as suggested by you to use a `ShortcodePreview` component. Also, I've addressed the other feedback comments you'd shared. Thanks for your time in reviewing and sharing feedback.

Reply:
Also, just a note - all tests have passed in my local system. Travis is having issues currently (I was told in #core-editor slack), hence the failure.